### ISSUE TITLE
masternodeConfig.getCount() always returns the number over 0

### ISSUE THREAD

**Issue by egonson** [dashpay/dash#1854]
https://github.com/dashpay/dash/blob/e596762ca22d703a79c6880a9d3edb1c7c972fd3/src/init.cpp#L1897

When I run dashd with no masternode, i got the error: "Enabling Masternode support requires turning on transaction indexing"

masternodeConfig.getCount() always returns 0 at initial time. so the if condition result is true.

I think masternodeConfig.getCount() returns vector's size() so the conditional expression must be masternodeConfig.getCount() > 0 not -1

**Comment by nmarley** [comment]
The code should be correct, but this looks like you have at least one entry in `masternode.conf` or you have set the `masternode=1` flag.

What do your `masternode.conf` and `dash.conf` entries look like? Also, are you starting dashd with the `masternode` flag?

**Comment by UdjinM6** [comment]
See https://github.com/dashpay/dash/pull/1817

tl;dr: Technically, it's a bug but we can't get rid of `txindex` for now, so it's always required, even in non-MN mode.

**Comment by nmarley** [comment]
@UdjinM6 In this case should we start ignoring the `-txindex` option altogether?

**Comment by UdjinM6** [comment]
I don't think that ignoring is a good option but I agree that providing some meaningful error message should help us to avoid confusions like this. Pls see #2014

**Comment by nmarley** [comment]
Closing as it's a duplicate of #1466 , and should be fixed with #2014 .
